### PR TITLE
The function :func:`efel.io.load_neo_file()` reads data from any of the file formats supported by Neo and formats it for use in eFEL.

### PR DESCRIPTION
**Pull request by jgonin**

### CODE REVIEW

**Comment by wvangeit** [comment]
Mmm, strange, this PR didn't trigger our continuous integration on Travis, will have to investigate this.

Btw, would you mind writing a test for your code ?

**Comment by wvangeit** [comment]
Ok, I found the issue. Was a wrong setting in Travis

**Comment by jgonin** [comment]
@wvangeit 
You are wellcome, I will make the tests today !

**Comment by jgonin** [comment]
@wvangeit 
Changes has been done, Travis is yelling because it doesn't know "neo"

**Comment by wvangeit** [comment]
Yes, I saw, thanks.
Regarding neo, I don't mind if you add it as a dependency in the setup.py of efel -> the 'install_requires' field.

However, I tried to run the tests with neo installed, and then I get another error:
======================================================================
ERROR: test_io.test_load_neo_file_stim_time_epoch
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/werner/local/pythonenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/werner/src/eFEL/efel/tests/test_io.py", line 164, in test_load_neo_file_stim_time_epoch
    result = efel.io.load_neo_file(file_name)
  File "/Users/werner/local/pythonenv/lib/python2.7/site-packages/efel/io.py", line 112, in load_neo_file
    blocks = reader.read()
  File "/Users/werner/local/pythonenv/lib/python2.7/site-packages/neo/io/baseio.py", line 116, in read
    return [self.read_block(lazy=lazy, cascade=cascade, **kargs)]
  File "/Users/werner/local/pythonenv/lib/python2.7/site-packages/neo/io/pickleio.py", line 45, in read_block
    block = pickle.load(fp)
AttributeError: 'module' object has no attribute '_new_epoch'

**Comment by jgonin** [comment]
Hum It might be because '_new_epoch' is a feature of the under development version of Neo which we added yesterday. The pickle files have been generated with the under development version of Neo and it would not work for the current release if an epoch is inside a pickle file. 

If you try to use the neo version from here https://github.com/NeuralEnsemble/python-neo I think it should work. If it is the case, then the next release of neo would fix this problem. 

Then I suggest I comment the test "test_load_neo_file_stim_time_epoch" for now until the new release of Neo. By the way, the function "reader.read()" would still be able to read epoch for other types of files.



"""
Regarding neo, I don't mind if you add it as a dependency in the setup.py of efel -> the 'install_requires' field.
"""
Ok then I will do it, thanks you.

File: efel/tests/neo_test_files/create_neo_files_test.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+rate = 1000*pq.Hz
+signal = neo.core.AnalogSignal(data, sampling_rate=rate, units="mV")
+seg.analogsignals.append(signal)
+seg.events.append(neo.core.Event(times=[0.0]*pq.ms, units=pq.ms, name="stim_start" ))
+seg.events.append(neo.core.Event(times=[20.0]*pq.ms, units=pq.ms, name="stim_end" ))
+bl.segments.append(seg)
+neo.io.PickleIO(filename=file_name, mode="rw").write_block(bl)
+
+#events incomplete
+file_name = "neo_test_file_events_time_incomplete.pickle"
```
Comment (mgeplf) [inline efel/tests/neo_test_files/create_neo_files_test.py]: would it be possible to not use pickles for the test? As you have seen, they are sensitive to class members, and are a leaky abstraction, which increases the cost of maintenance.

File: efel/io.py [hunk 2]
```diff
@@ -103,6 +103,7 @@ def load_neo_file (file_name, stim_start=None, stim_end=None):
             returned object : [Segments_1, Segments_2, ..., Segments_n] 
             Segments_1 = [Traces_1, Traces_2, ..., Traces_n] 
     """
+
```
Comment (mgeplf) [inline efel/io.py]: the load_neo_file is quite long, and factoring out the detection of start and end points would probably make it more clear.  That might also be useful code to have within the neo library, as it seems like common functionality that others would want (potentially?).

Comment (jgonin) [inline efel/io.py]: I think you are right, I will make these modifications. thanks you

File: efel/io.py [hunk 3]
```diff
@@ -126,33 +127,35 @@ def load_neo_file (file_name, stim_start=None, stim_end=None):
                 event_start = False
                 event_end = False
                 for event in seg.events :
-                    if event.name in ("stim_start", "stimulus_start", "stimulation_start", "current_injection_start") :   
-                        if stim_start is None :
-                            if event_start == False :
-                                event = event.rescale('ms').magnitude
-                                stim_start = event[0]
+                    rescaled = False
+                    if rescaled is False :
```
Comment (jgonin) [inline efel/io.py]: Yes, sorry, the "if rescaled is False :"at line 131  is a mistake.

File: efel/tests/neo_test_files/create_neo_files_test.py (hunk 1 above)
Comment (jgonin) [inline efel/tests/neo_test_files/create_neo_files_test.py]: Yes you are entirely right, it is done.

File: efel/io.py (hunk 2 above)
Comment (jgonin) [inline efel/io.py]: After consideration, there is no way we could make it usefull for Neo, it is quite specificly made for eFEL. Thus I will split it up as requested to increase the readability.

**Comment by jgonin** [comment]
Hello, I don't know codecov yet and it is not clear what I must do now. Could you advise me something here ?

**Comment by wvangeit** [comment]
Hi,
Codecov is complaining, because the code coverage as dropped below 100%.
The tests are not covering the following lines:
https://codecov.io/gh/BlueBrain/eFEL/src/467dd165fe541ad2a5e44a102b1a2d3f370986a6/efel/io.py

**Comment by wvangeit** [comment]
Changed a couple of small things, and merged.
Thanks again for your contribution

**Comment by jgonin** [comment]
It was a pleasure, you are welcome

**Comment by wvangeit** [comment]
Want to mention that 
https://github.com/BlueBrain/eFEL/pull/102
now adds neo as an 'extras' dependency.
Based on the comments in:
https://github.com/NeuralEnsemble/python-neo/issues/307

It means that if you want to use the neo functionality, you have to install eFEL using:
pip install efel[neo]
(or manually install neo and its dependencies)

**Pull request merged**